File: scrt/script/stptest.py
import sys
import os
import random

(strScriptPath, strScriptName) = os.path.split(__file__)
if strScriptPath not in sys.path:
	sys.path.insert(0, strScriptPath)
sys.dont_write_bytecode = True

import session

sess = session.sess(crt.GetActiveTab())
loopcmdnum = 5 + random.randint(0,9)
timeout = 5 + random.randint(0,9)
cmdarray = (
	"interface eth0/25\r\nshutdown\r\nexit",
	"interface eth0/25\r\nno shutdown\r\nexit",
	"interface eth0/26\r\nshutdown\r\nexit",
	"interface eth0/26\r\nno shutdown\r\nexit",
	"interface eth0/27\r\nshutdown\r\nexit",
	"interface eth0/27\r\nno shutdown\r\nexit",
	"interface eth0/28\r\nshutdown\r\nexit",
	"interface eth0/28\r\nno shutdown\r\nexit"
)

def	execcmd(cmdstr): 
	crt.Screen.Send(cmdstr + "\n")

def	cmdpreconfig(): 
	crt.Screen.Send("enable\nconfig t\n")
	return 1

def	cmdloop(num): 
	while num>0 and not sess.wait(timeout):
		num = num - 1
		if "Terminating" in sess.get_output(): # exception when waiting
			return False
		if not sess.cmdsexec(cmdarray[random.randint(0,len(cmdarray) - 1)]):
			return False
		if "Terminating" in sess.get_output():
			return False
	return (num == 0)

crt.Screen.Send("#loopcmdnum " + str(loopcmdnum) + " timeout " + str(timeout) + "\n")
while cmdloop(loopcmdnum) and sess.cmdreboot() and sess.wait2login():pass
crt.Screen.Send("#game over!\n")

File: scrt/script/stptest.js
var loopcmdnum = 5 + Math.floor(Math.random() * 10);
var timeout = 5 + Math.floor(Math.random() * 10);
var prompt1 = "login:";
var prompt2 = "RETURN"; // unused
var intr1 = "^C";
var intr2 = "<INTERRUPT>";

var cmdarray = new Array(
  "interface eth0/7\r\nshutdown\r\nexit",
  "interface eth0/7\r\nno shutdown\r\nexit",
  "interface eth0/8\r\nshutdown\r\nexit",
  "interface eth0/8\r\nno shutdown\r\nexit"
);

function execcmd(cmdstr) {
  crt.Screen.Send(cmdstr + "\n");
}

function wait4pause(time) {
  var ret = crt.Screen.WaitForStrings("^C", "<INTERRUPT>", time);
  if (ret > 0) {
    crt.Screen.Send("###user terminated(" + ret + ")!\n");
    return 1;
  }
  return 0;
}

function cmdreboot() {
  crt.Screen.Send("end\nentershell\n");
  crt.Screen.Send("reboot\n");
  return 1;
}

function cmdpreconfig() {
  crt.Screen.Send("enable\nconfig t\n");
  return 1;
}

function cmdwait2login() {
  var ret = crt.Screen.WaitForStrings(prompt1, prompt2, intr1, intr2);
  if (ret == 1 || ret == 2) {
    crt.Screen.Send("admin\n");
    crt.Sleep(500);
    crt.Screen.Send("admin\n");
    crt.Sleep(1000);
    return !wait4pause(5) && cmdpreconfig(); // wait for a moment, make sure system started.
  } else {
    return 0;
  }
}

function cmdloop(num) {
  while (num-- && !wait4pause(timeout)) {
    execcmd(cmdarray[Math.floor(Math.random() * cmdarray.length)]);
  }
  return num == -1;
}

crt.Screen.send("#loopcmdnum " + loopcmdnum + " timeout " + timeout + "\n");
while (cmdloop(loopcmdnum) && cmdreboot() && cmdwait2login());
crt.Screen.Send("#game over!\n");
